controller/rest/system.go

```diff
@@ -2116,6 +2117,223 @@ func handlerSystemGetRBAC(w http.ResponseWriter, r *http.Request, ps httprouter.
 	restRespSuccess(w, r, &resp, acc, login, nil, "Get missing Kubernetes RBAC")
 }
 
+func getAlertGroup(alerts []string, alertType api.AlertType, acceptedAlerts utils.Set) (*api.RESTNvAlertGroup, error) {
+	alertGroup := &api.RESTNvAlertGroup{
+		Type: alertType,
+	}
+
+	if len(alerts) > 0 {
+		for _, alert := range alerts {
+			b := md5.Sum([]byte(alert))
+			key := hex.EncodeToString(b[:])
+			if !acceptedAlerts.Contains(key) {
+				alertGroup.Data = append(alertGroup.Data, &api.RESTNvAlert{
+					ID:      key,
+					Message: alert,
+				})
+			}
+		}
+	}
+
+	if len(alertGroup.Data) > 0 {
+		return alertGroup, nil
+	}
+
+	return alertGroup, errors.New("no alert is retrieved")
+}
+
+func getInternalCertExpireAlert(certFilePath string) (string, error) {
+	expireThresholds := []time.Duration{30 * 24 * time.Hour, 90 * 24 * time.Hour, 180 * 24 * time.Hour}
+	// Check ca certificate expiration
+	for i, expireThreshold := range expireThresholds {
+		certExpired, err := IsCertNearExpired(certFilePath, expireThreshold)
+		if err != nil {
+			return "", err
+		}
+		if !certExpired {
+			continue
+		}
+
+		var certExpiredMsg string
+		switch i {
+		case 0:
+			if strings.Contains(certFilePath, "ca.cert") {
+				certExpiredMsg = "Internal CA certificate will be expired in 30 days."
+			} else {
+				certExpiredMsg = "Internal certificate will be expired in 30 days."
+			}
+		case 1:
+			if strings.Contains(certFilePath, "ca.cert") {
+				certExpiredMsg = "Internal CA certificate will be expired in 90 days."
+			} else {
+				certExpiredMsg = "Internal certificate will be expired in 90 days."
+			}
+		case 2:
+			if strings.Contains(certFilePath, "ca.cert") {
+				certExpiredMsg = "Internal CA certificate will be expired in 180 days."
+			} else {
+				certExpiredMsg = "Internal certificate will be expired in 180 days."
+			}
+		}
+		return certExpiredMsg, nil
+	}
+
+	return "", errors.New("no nearly expired internal certificate is detected")
```

Comment:
Maybe it's better to consider the case where the code runs successfully but no alert is found as a success?

Reply:
changed as above